docs/source/perf/spmd_basic.md

```diff
@@ -61,56 +65,133 @@ SPMD with other distributed libraries.
 
 ### Mesh
 
-For a given cluster of devices, a physical mesh is a representation of
-the interconnect topology.
+The SPMD programming model is built around the concept of a device mesh, commonly
+referred to as a mesh. A device mesh is a logical N-dimensional arrangement of
+compute devices (e.g. TPU cores) where the axes are used as shorthands to
+define sharded computation. It is logical in the sense that the mesh shape does
+not necessarily have to reflect the physical network layout. You could create
```

Comment:
You can create differently shaped meshes over the same set of physical devices.

Reply:
done